src/natcap/invest/ndr/ndr.py

```diff
@@ -996,6 +990,42 @@ def _mult_op(*array_nodata_list):
         gdal.GDT_Float32, target_nodata)
 
 
+def _sum_rasters(raster_path_list, target_nodata, target_result_path):
+    """Sum the rasters in `raster_path_list`.
+
+    Args:
+        raster_path_list (list): list of single band raster paths.
+        target_nodata (float): desired target nodata value.
+        target_result_path (string): path to float 32 target raster
+            multiplied where all rasters are not nodata.
+
+    Returns:
+        None.
+
+    """
+    def _sum_op(*array_nodata_list):
+        """Multiply non-nodata stacks."""
+        result = numpy.empty(array_nodata_list[0].shape)
+        result[:] = target_nodata
+        valid_mask = numpy.full(result.shape, True)
+        for array, nodata in zip(*[iter(array_nodata_list)]*2):
```

Comment:
Could you walk me through `*[iter(array_nodata_list)]*2`? I'm having a hard time understanding how it's unpacking with `zip`? Thanks!

Reply:
I just copied all this from the similar raster multiplication function in NDR haha. I should've looked more closely - this implementation is pretty confusing. Refactored the whole thing.